pkg/account/api/search_filter.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 The Bucketeer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"context"
+	"errors"
+
+	"go.uber.org/zap"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+
+	"github.com/bucketeer-io/bucketeer/pkg/account/command"
+
+	"github.com/bucketeer-io/bucketeer/pkg/account/domain"
+
+	v2as "github.com/bucketeer-io/bucketeer/pkg/account/storage/v2"
+	"github.com/bucketeer-io/bucketeer/pkg/locale"
+	"github.com/bucketeer-io/bucketeer/pkg/log"
+	accountproto "github.com/bucketeer-io/bucketeer/proto/account"
+)
+
+func (s *AccountService) CreateSearchFilterV2(
+	ctx context.Context,
+	req *accountproto.CreateSearchFilterRequest,
+) (*accountproto.CreateSearchFilterResponse, error) {
+	localizer := locale.NewLocalizer(ctx)
+	editor, err := s.checkEnvironmentRole(
+		ctx, accountproto.AccountV2_Role_Environment_UNASSIGNED,
+		req.EnvironmentNamespace, localizer)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := validateCreateSearchFilterRequest(req, localizer); err != nil {
+		s.logger.Error(
+			"Failed to create search filter",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		return nil, err
+	}
+
+	account, err := s.getAccountV2(ctx, req.Email, req.OrganizationId, localizer)
+	if err != nil {
+		return nil, err
+	}
+	// Since there is only one default setting for a filter target, set the existing default to OFF.
+	changeDefaultFilters := getChangeDefaultFilters(account, req.Command.SearchFilter)
```

Comment:
It's better not to change the default setting for existing filters. When creating a new filter, we don't necessarily want to be the default setting.

Reply:
@cre8ivejp 

> When creating a new filter, we don't necessarily want to be the default setting.

As you said, I know there is no need to set default settings at creation time.

The process here is to avoid having multiple default filters.

So, if you already have a filter with default settings and you want to set the newly created filter as the default, I think you need to turn off the previous default settings. What do you think?